src/ResourceManager/Advisor/Commands.Advisor/Cmdlets/Utilities/RecommendationHelper.cs

```diff
@@ -0,0 +1,196 @@
+                        if (entry.Category.Equals(category))
+                        {
+                            filteredList.Add(entry);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(resourceGroup))
+                {
+                    foreach (PsAzureAdvisorResourceRecommendationBase entry in recListTobeFiltered)
+                    {
+                        if (entry.Id.Contains(resourceGroup))
+                        {
+                            filteredList.Add(entry);
+                        }
+                    }
+                }
+            }
+            return filteredList;
+        }
+
+        /// <summary>
+        /// Filter recommendations by given recommendationName.
+        /// </summary>
+        /// <param name="recListTobeFiltered">List to be filtered</param>
+        /// <param name="recommendationName">Name of the recommendation, type GUID.</param>
+        /// <returns>Recommendation of PsAzureAdvisorResourceRecommendationBase type</returns>
+        public static PsAzureAdvisorResourceRecommendationBase RecomendationFilterByRecommendation(IEnumerable<PsAzureAdvisorResourceRecommendationBase> recListTobeFiltered, string recommendationName)
+        {
+            foreach (PsAzureAdvisorResourceRecommendationBase recommendationEntry in recListTobeFiltered)
+            {
+                if (recommendationEntry.Name.Equals(recommendationName))
+                {
+                    return recommendationEntry;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Filter recommendations by given recommendationNameList.
+        /// </summary>
+        /// <param name="recListTobeFiltered">List to be filtered</param>
+        /// <param name="recommendationNameList">List of recommendation-names, type GUID.</param>
+        /// <returns>Recommendation list of PsAzureAdvisorResourceRecommendationBase type</returns>
+        public static List<PsAzureAdvisorResourceRecommendationBase> RecomendationFilterByRecommendation(IEnumerable<PsAzureAdvisorResourceRecommendationBase> recListTobeFiltered, List<string> recommendationNameList)
+        {
+            List<PsAzureAdvisorResourceRecommendationBase> returnList = new List<PsAzureAdvisorResourceRecommendationBase>();
+            foreach (string recommendaitonName in recommendationNameList)
+            {
+                returnList.Add(RecomendationFilterByRecommendation(recListTobeFiltered, recommendaitonName));
+            }
+
+            return returnList;
+        }
+
+        /// <summary>
+        /// Parse the subscriptionId from the resourceId.
+        /// </summary>
+        /// <param name="resourceID">ResourceId of recommendation</param>
+        /// <returns>SubscriptionId as string</returns>
+        public static string GetSubscriptionIdfromResoureID(string resourceID)
```

Comment:
use this instead: https://github.com/Azure/azure-powershell-common/blob/master/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs

Reply:
Removed it.